docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,270 @@
+
+### How to implement PCA on an existing indexer?
+It's the common case that we have limited data at the first time but then more and more data come in. So how to implement PCA on an existing indexer?
+
+We still have to prepare training data for PCA, but now the training data come from indexer files directly:
+```python
+from annlite.index import AnnLite
+import numpy as np
+
+
+index = AnnLite(dim=512, n_components=128)
+train_data = index.load_embeddings(limit=10e5)
+index.train(train_data)
+```
+
+`index.load_embeddings()` will load embeddings from `lmdb` with size of limit.
+
+
+## How to deal with a very large dataset?
+For a very large dataset, for example, 100 million data or even 1 billion data, it's not possible to implement index operations on a single machine. **Sharding**, a type of partitioning that separates a large dataset into smaller, faster, more easily managed parts, is needed in this case.
+
+You need to specify the `shards` and `polling` in the YAML config:
+```yaml
+jtype: Flow
+version: '1'
+with:
+  port: 51000
+executors:
+  - name: encoder
+    uses:
+      jtype: CLIPEncoder
+      metas:
+        py_modules:
+          - clip_server.executors.clip_torch
+  - name: indexer
+  	uses:
+  	  jtype: AnnLiteIndexer
+  	  with:
+  	  	dim: 512  # input dimension of PCA
+  	  	n_components: 128  # output dimension of PCA
+  	  metas:
+  	  	py_modules:
+  	  	  - annlite.executor
+  	shards: 5
+  	polling: {'/index': 'ANY', '/search': 'ALL', '/update': 'ALL',
+              '/delete': 'ALL', '/status': 'ALL'}
+```
+
+| Parameter               | Description                                                                                                                  |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| `shards`                  | Number of shardings. |
+| `polling` | Polling strategies for different endpoints.|
+
+Then you can perform exactly the same operations as we do on a single machine.(`/encode`, `/index` and `/search`)
+
+### Why different [polling strategies](https://docs.jina.ai/how-to/scale-out/?highlight=polling#different-polling-strategies) are needed for different endpoints?
+
+Differences between `ANY` and `ALL`:
+- `ANY`: requests are sent to one of the executors.
+- `ALL`: requests are sent to all executors.
+
+```{figure} images/polling_stratey.png
+:width: 80%
+
+```
+
+Since one data point only needed to be indexed once, there will only be one indexer executor that will handle this data point. Thus, `ANY` is used for `/index`. On the contrary, search operations needed to be handled by all indexer executors since we don't know which executor store the perfectly matched result.
```

Comment:
```suggestion
Since one data point only needs to be indexed once, there will only be one indexer executor that will handle this data point. Thus, `ANY` is used for `/index`. On the contrary, search operations needed to be handled by all indexer executors since we don't know which executor stores the perfectly matched result so `ALL` is used.
```